src/django/pfb_analysis/views.py

```diff
@@ -0,0 +1,48 @@
+from rest_framework.filters import DjangoFilterBackend, OrderingFilter
+from rest_framework.viewsets import ModelViewSet
+
+from pfb_analysis.models import AnalysisJob, Neighborhood
+from pfb_analysis.serializers import AnalysisJobSerializer, NeighborhoodSerializer
+from pfb_network_connectivity.filters import OrgAutoFilterBackend, SelfUserAutoFilterBackend
+from pfb_network_connectivity.permissions import RestrictedCreate
+from users.models import PFBUser
+
+
+class AnalysisJobViewSet(ModelViewSet):
+    """
+    For listing or retrieving analysis jobs.
+    """
+    queryset = AnalysisJob.objects.all()
+    serializer_class = AnalysisJobSerializer
+    permission_classes = (RestrictedCreate,)
+    filter_fields = ('status', 'neighborhood',)
+    filter_backends = (DjangoFilterBackend, OrderingFilter,
+                       OrgAutoFilterBackend, SelfUserAutoFilterBackend)
+    ordering_fields = ('created',)
+
+    def perform_create(self, serializer):
+        """ Start analysis jobs as soon as created """
+        instance = serializer.save()
+        instance.run()
+
+    def perform_update(self, serializer):
```

Comment:
It seems to me that we should make AnalysisJob immutable (to the user) once its been created. If a particular job needs to be re-run it can be submitted as a new job with the same parameters. Otherwise we potentially destroy information if the older run was successful.